plugins/gcov/gcov.rake

```diff
@@ -432,6 +483,116 @@ namespace UTILS_SYM do
+  def report_generator_args_builder(opts)
+    rg_opts = get_report_generator_opts(opts)
+
+    args = ""
+    args += "\"-reports:*.gcov\" "
+    args += "\"-targetdir:\"#{GCOV_REPORT_GENERATOR_PATH}\"\" "
+
+    # Build the report types argument.
+    if !(opts.nil?) && !(opts[:gcov_reports].nil?) && !(opts[:gcov_reports].empty?)
+      args += "\"-reporttypes:"
+
+      for report_type in opts[:gcov_reports]
+        rg_report_type = REPORT_TYPE_TO_REPORT_GENERATOR_REPORT_NAME[report_type.upcase]
+        if !(rg_report_type.nil?)
+          args += rg_report_type + ";"
+        end
+      end
+
+      # Removing trailing ';' after the last report type.
+      args = args.chomp(";")
+
+      # Append a space seperator after the report type.
+      args += "\" "
+    end
+
+    # Build the source directories argument.
+    args += "\"-sourcedirs:.;"
+    if !(opts[:collection_paths_source].nil?)
+      for src_path in opts[:collection_paths_source]
+        args += src_path + ";"
+      end
+    end
+    args = args.chomp(";")
+    args += "\" "
+
+    args += "\"-historydir:#{rg_opts[:history_directory]}\" " unless rg_opts[:history_directory].nil?
+    args += "\"-plugins:#{rg_opts[:plugins]}\" " unless rg_opts[:plugins].nil?
+    args += "\"-assemblyfilters:#{rg_opts[:assembly_filters]}\" " unless rg_opts[:assembly_filters].nil?
+    args += "\"-classfilters:#{rg_opts[:class_filters]}\" " unless rg_opts[:class_filters].nil?
+    args += "\"-filefilters:#{rg_opts[:file_filters] || GCOV_REPORT_GENERATOR_FILE_FILTERS}\" "
+    args += "\"-verbosity:#{rg_opts[:verbosity] || "Warning"}\" "
+    args += "\"-tag:#{rg_opts[:tag]}\" " unless rg_opts[:tag].nil?
+
+    return args
+  end
+
+
+  # Run gcov with the given arguments.
+  def run_gcov(args)
+    command = @ceedling[:tool_executor].build_command_line(TOOLS_GCOV_GCOV_POST_REPORT, [], args)
+    return @ceedling[:tool_executor].exec(command[:line], command[:options])
+  end
+
+
+  # Run Report Generator with the given arguments.
+  def run_report_generator(args)
+    command = @ceedling[:tool_executor].build_command_line(TOOLS_GCOV_REPORT_GENERATOR_POST_REPORT, [], args)
+    return @ceedling[:tool_executor].exec(command[:line], command[:options])
+  end
+
+
+  # Generate the Report Generator report(s) specified in the options.
+  def make_report_generator_reports(opts)
+    shell_result = nil
+    total_time = Benchmark.realtime do
+      rg_opts = get_report_generator_opts(opts)
+
+      print "Creating gcov results report(s) with Report Generator in '#{GCOV_REPORT_GENERATOR_PATH}'... "
+      STDOUT.flush
+
+      # Cleanup any existing .gcov files to avoid reporting old coverage results.
+      for gcov_file in Dir.glob("*.gcov")
+        File.delete(gcov_file)
+      end
+
+      # Avoid running gcov on the mock, test, and unity gcov notes files to save time.
+      gcno_exclude_regex = /(\/|\\)(#{opts[:cmock_mock_prefix]}.*|#{opts[:project_test_file_prefix]}.*|unity|cmock)\.gcno/
```

Comment:
Can we use VENDORS_FILES from constants.rb instead of `...|unity|cmock)`?

Reply:
Good idea. I made the change.